src/pdm/cli/commands/run.py

```diff
@@ -216,12 +228,24 @@ def _run_task(
         kind, _, value, options = task
         shell = False
         if kind == "cmd":
-            if not isinstance(value, list):
-                value = shlex.split(str(value))
-            args = value + list(args)
+            interpolations = 0
+            if isinstance(value, str):
+                cmd, count = interpolate(value, args)
+                interpolations += count
+                value = shlex.split(cmd)
+            else:
+                agg = [interpolate(part, args) for part in value]
+                interpolations += sum(row[1] for row in agg)
```

Comment:
```suggestion
                interpolations = sum(row[1] for row in agg)
```

Reply:
I refactored ² interpolate²  to return `tuple[str, bool]` and removed the need for loop and aggregate.
It should be cleaner and easier to read and understand